packages/concerto-core/test/decoratormanager.js

```diff
@@ -470,4 +470,32 @@ describe('DecoratorManager', () => {
             }).should.throw(/Model violation in the "concerto.metamodel@1.0.0.Decorator" instance. Invalid enum value of "INVALID" for the field "CommandType"/);
         });
     });
-});
+
+    describe('#extractDecorators', function() {
+        it('should be able to extract decorators and vocabs from a model', async function() {
+            const testModelManager = new ModelManager({strict:true});
+            const modelText = fs.readFileSync('./test/data/decoratorcommands/test-decorated-model.cto', 'utf-8');
+            testModelManager.addCTOModel(modelText, 'test.cto');
+            const options = {
+                removeDecoratorsFromModel:true,
+                locale:'en'
+            };
+            let resp = DecoratorManager.extractDecorators( testModelManager, options);
+            resp.should.not.be.null;
```

Comment:
Can we be more precise about what the response should be, please?

Reply:
You could also confirm that the response is a valid decorator set? Furthermore, I assume that extracting/removing decorators and then reapplying them should be an identity operation?